### PR TITLE
V2 issue703

### PR DESCRIPTION
**Pull request by petersilva**
change Integrity to identity in v03 format messages. implement issue #703 for v2 code base.
needs to match sr3 and C implementations.

### CODE REVIEW

**Comment by petersilva** [comment]
oops. already merged in C 929b6f7d81835a0395162045ab1df6d15b4aa0c6 forgot to do a PR.

**Comment by petersilva** [comment]
I'm pretty sure the dynamic flow tests have nothing to do with the changes made... the failures were already present, and with the same symptoms prior to this pull request.

probably best to proceed.   I'm looking at the resolving the problem with the dynamic_flows, but it is a separate issue.

File: sarra/sr_message.py [hunk 1]
```diff
@@ -296,18 +296,23 @@ def from_amqplib(self, msg=None ):
...
+
+               if "identity" in self.headers.keys():
+                   if type( self.headers[ "identity" ] ) is str:
+                       self.headers[ "identity" ] = json.loads( self.headers[ "identity" ] )
+                   sa = sum_algo_v3tov2[ self.headers[ "identity" ][ "method" ] ]
 
                    # transform sum value
                    if sa in [ '0' ]:
-                       sv = self.headers[ "integrity" ][ "value" ]
+                       sv = self.headers[ "indentity" ][ "value" ]
```
Comment (reidsunderland) [inline sarra/sr_message.py:311]: typo! :) `indentity` should be `identity`

File: sarra/sr_xattr.py [hunk 2]
```diff
@@ -127,6 +127,10 @@ def __init__(self, path):
                 #   unlikely to be a difficulty in real life? but perhaps worthy of an error message.
                 pass
 
+        if 'integrity' in self.x:
```
Comment (reidsunderland) [inline sarra/sr_xattr.py:130]: same as before, in sr_message.py. Not sure if it matters

**Review by reidsunderland — CHANGES_REQUESTED** [review]
I left a few comments. There's one typo that needs to be fixed and a question about what to do if somehow a message is received with both identity and integrity headers.

File: sarra/sr_message.py [hunk 3]
```diff
@@ -296,18 +296,23 @@ def from_amqplib(self, msg=None ):
                self.baseurl = self.headers[ "baseUrl" ]
                self.relpath = self.headers[ "relPath" ]
                self.notice = "%s %s %s" % ( self.pubtime, self.baseurl, self.relpath )
+
                if "integrity" in self.headers.keys():
```
[the remark this answers is not in the document]
Comment (petersilva) [reply sarra/sr_message.py:300]: I think of this as a transitional thing... by which I mean that, in a year or two, the support for "integrity" ought to be removed. In the mean-time faced with this ambiguity, I don't see a clearly correct behaviour.

maybe it should be tagged as such ....(A comment like "remove this in 2025 ?" )dunno.

File: sarra/sr_xattr.py (hunk 2 above)
Comment (petersilva) [reply sarra/sr_xattr.py:130]: that one is correct. it's to "upgrade" attributes when written by a previous version. convert it to identity
within the code.

**Review by reidsunderland — APPROVED** [review]

**Pull request merged**